client/state/selectors/get-current-query-arguments.ts

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable jsdoc/require-param */
+
+/**
+ * Internal dependencies
+ */
+import 'calypso/state/route/init';
+
+/**
+ * Type dependencies
+ */
+import type { Query } from 'calypso/state/route/types';
+import type { IAppState } from 'calypso/state/types';
+
+/**
+ * Gets the last query parameters set by a ROUTE_SET action
+ *
+ * TODO: We would be better off not conditioning with null here.
+ * Kept for backwards compatibility in case any null checks.
+ * Investigate and refactor.
+ */
+export function getCurrentQueryArguments( state: IAppState ): Query | false | null {
```

Comment:
The return type should be just `Query`, and the selector should return an empty `{}` object if the current URL doesn't have a query string. That's the API most convenient for the caller.

Reply:
I feel the selector could just return whatever `state.route.query.current` points to. 🤔 No need to coalesce `null` in the mix as we do here either (as also done previously). So either `undefined` or `state.route.query.current` -> ultimately, just `return state?.route?.query?.current;`. 

This makes sense to me in the following way: 
- the selector becomes a simple/dumb reflection of the state tree
- the default `{}` (or whatever else we choose) becomes the responsibility of the reducer
- undefined for something that has not been defined yet may carry meaning of its own, and a raw/base selector can reflect that

Of course, trade-offs e.g. a derived state selector will ultimately provide some sort of default, which they'd be responsible for defining anyhow. So the alternative might just be better e.g. selectors always defining and returning the default values.

Reply:
Yes, I think the `false` value for the state doesn't need to be there, in neither of `initial`, `current` and `previous`. It can be an empty object instead:
```js
const EMPTY_QUERY = {};

const initialState = {
  initial: EMPTY_QUERY,
  current: EMPTY_QUERY,
  previous: EMPTY_QUERY,
};
```
And the reducer for `initial` can do a check like:
```js
initial = state.initial === EMPTY_QUERY ? action.query : state.initial
```
to update state only on the very first `ROUTE_SET`. Even if `action.query` will be an empty object, it will be a _different_ empty object than the initial state.

Reply:
That's perfect. Thanks! We'll take this path.